File: transportable-udfs-codegen/src/test/resources/outputs/sample-udf-properties/hive/sources/udfs/hive/SimpleUDF.java
package udfs.hive;

import com.google.common.collect.ImmutableList;
import com.linkedin.transport.api.udf.StdUDF;
import com.linkedin.transport.api.udf.TopLevelStdUDF;
import com.linkedin.transport.hive.StdUdfWrapper;
import java.lang.Class;
import java.lang.Override;
import java.util.List;

public class SimpleUDF extends StdUdfWrapper {
  @Override
  protected Class<? extends TopLevelStdUDF> getTopLevelUdfClass() {
    return udfs.SimpleUDF.class;
  }

  @Override
  protected List<? extends StdUDF> getStdUdfImplementations() {
    return ImmutableList.of(new udfs.SimpleUDF());
  }
}

File: transportable-udfs-codegen/src/test/resources/outputs/sample-udf-properties/spark/sources/udfs/spark/SimpleUDF.scala
package udfs.spark

import java.util
import com.google.common.collect.ImmutableList
import com.linkedin.transport.api.udf.{StdUDF, TopLevelStdUDF}
import com.linkedin.transport.spark.StdUdfWrapper
import org.apache.spark.sql.catalyst.expressions.Expression


case class SimpleUDF(expressions: Seq[Expression]) extends StdUdfWrapper(expressions) {

  override protected def getTopLevelUdfClass: Class[_ <: TopLevelStdUDF] = classOf[udfs.SimpleUDF]

  override protected def getStdUdfImplementations: util.List[_ <: StdUDF] = ImmutableList.of(
    new udfs.SimpleUDF()
  )
}
